low-level/framework/utils/consulstore.py

```diff
@@ -93,48 +141,76 @@ def get(self, key, **kwargs):
             except requests.exceptions.ConnectionError as connerr:
                 logger.warn("Error[{0}] consul connection refused Retry Index {1}" \
                     .format(connerr, retry_index))
+                self._consul_conn_status = False
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
                 logger.warn("Error[{0}] while reading data from consul {1}" \
-                    .format(gerr, key))
+                    .format(gerr, consul_key))
+                self._consul_conn_status = False
                 break
+
+        return data
+
+    def get(self, key, **kwargs):
+        """Load data from given key."""
+        data = None
+        if self._consul_conn_status is False:
+            self._establish_consul_connection(self._host, self._port)
+
+        if self._consul_conn_status and self._data_sync_required:
+            self._dump_filestore_to_consulstore()
```

Comment:
Add a TODO to make this asynchronous task, trigger can happen here but current get() should not get delayed

Reply:
done